tests/phpunit/tests/date/wpDate.php

```diff
@@ -61,4 +61,118 @@ public function test_should_keep_localized_slashes() {
 
 		$this->assertSame( $string, wp_date( 'F', $datetime->getTimestamp(), $utc ) );
 	}
+
+	/**
+	 * Tests that the date is formatted with no timestamp provided.
+	 *
+	 * @ticket 53485
+	 */
+	public function test_should_format_date_with_no_timestamp() {
+		$utc = new DateTimeZone( 'UTC' );
+		$this->assertSame( (string) time(), wp_date( 'U', null, $utc ) );
+	}
+
+	/**
+	 * Tests that the date is formatted with no timezone provided.
+	 *
+	 * @ticket 53485
+	 */
+	public function test_should_format_date_with_no_timezone() {
+		$utc      = new DateTimeZone( 'UTC' );
+		$datetime = new DateTimeImmutable( '2019-10-17', $utc );
+		$this->assertSame( 'October', wp_date( 'F', $datetime->getTimestamp() ) );
+	}
+
+	/**
+	 * Tests that the format is set correctly.
+	 *
+	 * @dataProvider data_should_format_date
+	 *
+	 * @ticket 53485
+	 *
+	 * @param string $expected The expected result.
+	 * @param string $format   The date format.
+	 */
+	public function test_should_format_date( $expected, $format ) {
+		$utc      = new DateTimeZone( 'UTC' );
+		$datetime = new DateTimeImmutable( '2019-10-17', $utc );
+
+		$this->assertSame( $expected, wp_date( $format, $datetime->getTimestamp(), $utc ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_should_format_date() {
+		return array(
+			'Swatch Internet Time'                        => array(
+				'expected' => '041',
+				'format'   => 'B',
+			),
+			'Ante meridiem and Post meridiem (uppercase)' => array(
+				'expected' => 'AM',
+				'format'   => 'A',
+			),
+			'Ante meridiem and Post meridiem (uppercase) and escaped "A"' => array(
+				'expected' => 'A AM',
+				'format'   => '\\A A',
+			),
+			'Ante meridiem and Post meridiem (lowercase)' => array(
+				'expected' => 'am',
+				'format'   => 'a',
+			),
+			'Month'                                       => array(
+				'expected' => 'October',
+				'format'   => 'F',
+			),
+			'Month (abbreviated'                          => array(
+				'expected' => 'Oct',
+				'format'   => 'M',
+			),
+			'Weekday'                                     => array(
+				'expected' => 'Thursday',
+				'format'   => 'l',
+			),
+			'Weekday (abbreviated)'                       => array(
+				'expected' => 'Thu',
+				'format'   => 'D',
+			),
+		);
+	}
+
+	/**
+	 * Tests that the date is formatted when
+	 * `$wp_locale->month` and `$wp_locale->weekday` are empty.
+	 *
+	 * @ticket 53485
+	 */
+	public function test_should_format_date_with_empty_wp_locale_month_and_weekday() {
+		global $wp_locale;
+
+		$utc      = new DateTimeZone( 'UTC' );
+		$datetime = new DateTimeImmutable( '2019-10-17', $utc );
+
+		$original_wp_locale = $wp_locale;
+		$wp_locale->month   = array();
+		$wp_locale->weekday = array();
+		$actual             = wp_date( 'F', $datetime->getTimestamp(), $utc );
+		$wp_locale          = $original_wp_locale;
```

Comment:
```suggestion
		$wp_locale->month   = array();
		$wp_locale->weekday = array();
		$actual             = wp_date( 'F', $datetime->getTimestamp(), $utc );
```
$global can go from start of method too.



setUp and tearDown look to handle this. 

Correct me if I am misunderstanding your purpose here, in which case you'll need to add a `clone`.

Reply:
👍 You're right. I hadn't checked `setUp` and `tearDown`. Woops!

Reply:
> Is it possible for the test suite to run in an unexpected timezone and mess up the expectations here (for example an Aussie timezone woulld switch am to pm, etc). If possible, then it is probably worth setting the timezone for the tests.

Hmm, I'm actually not sure whether or not that's possible. Do they fail for you locally, for example?

Reply:
> Do they fail for you locally, for example?

They're working for me using the default tests config, I just noticed that other date related tests set the `timezone_string` option so figured I'd ask.

For the sake of certainty, it's probably worth including `update_option( 'timezone_string', 'UTC' );` in the set up. It looks to be fairly common in other classes testing date functions.

I can never remember if options and reset between each test and/or class tear down.

Reply:
I ran two test methods:
1. Set the `home` option to `http://something` and verify it was set. ✅
2. Check if `home` is still `http://something`. ❌

So the options are reset between test methods. 🎉